### PR TITLE
Drop hourly partitions when daily data gets converted to ORC

### PR DESCRIPTION
**Pull request by pcadabam**
@abti can you review?

### CODE REVIEW

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java [hunk 1]
```diff
@@ -824,6 +828,33 @@ public static String generateCleanupDDL(String stagingTableName, Optional<String
...
+      return StringUtils.EMPTY;
+    }
+
+    StringBuilder ddl = new StringBuilder(String.format("ALTER TABLE %s DROP IF EXISTS ", finalTableName));
+
+    // Join the partition specs
+    ddl.append(Joiner.on(", ").join(Iterables.transform(partitionDMLInfos, PARTITION_SPEC_GENERATOR)));
+
+    if (purge) {
+      ddl.append(" PURGE");
```
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java]: Purge is not available in 0.13, also it only purges data for managed Hive tables not external. So we need to manually cleanup data via FS apis

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java [hunk 2]
```diff
@@ -824,6 +828,33 @@ public static String generateCleanupDDL(String stagingTableName, Optional<String
...
+   * @param partitionDMLInfos list of Partition to be dropped
+   * @return DDL to drop partitions in <code>finalTableName</code>
+   */
+  public static String generateDropPartitionsDDL(final String finalTableName, final List<Map<String, String>> partitionDMLInfos, boolean purge) {
+
+    if (partitionDMLInfos.isEmpty()) {
+      return StringUtils.EMPTY;
+    }
+
+    StringBuilder ddl = new StringBuilder(String.format("ALTER TABLE %s DROP IF EXISTS ", finalTableName));
```
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java]: Please specify 'USE {db}' too for safety before alter

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java [hunk 3]
```diff
@@ -824,6 +828,33 @@ public static String generateCleanupDDL(String stagingTableName, Optional<String
...
+
+    // Join the partition specs
+    ddl.append(Joiner.on(", ").join(Iterables.transform(partitionDMLInfos, PARTITION_SPEC_GENERATOR)));
+
+    if (purge) {
+      ddl.append(" PURGE");
+    }
+    ddl.append(";");
```
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/query/HiveAvroORCQueryGenerator.java]: Remove semi-colon, Hive Jdbc doesnt likes it

**Comment by pcadabam** [comment]
@abti addressed all comments

**Comment by abti** [comment]
LGTM

**Pull request merged**